tests/driver/Firestore/InProcessFirestore.set.test.ts

```diff
@@ -144,6 +144,29 @@ describe("InProcessFirestore set", () => {
             expect(stored.exists).toBe(false)
             expect(stored.data()).toEqual({})
         })
+
+        test("throws with an undefined nested field", async () => {
+            // When we set a new document in a collection, it throws;
+            await expect(
+                firestore
+                    .collection("myCollection")
+                    .doc("thing")
+                    .set({
+                        name: "thing",
+                        good: {
+                            thing: undefined,
+                        },
+                    }),
+            ).rejects.toThrowError(FirestoreError)
```

Comment:
message: "Value for argument "data" is not a valid Firestore document. Cannot use "undefined" as a Firestore value (found in field "body.`0`.account.legalEntity"). If you want to ignore undefined values, enable `ignoreUndefinedProperties`." - can we get something that resembles this?

Reply:
The message is pretty much that: "Value for argument "data" is not a valid Firestore document. Cannot use "undefined" as a Firestore value (found in fields ${undefinedFields})"